src/MentionsTextInput.js

```diff
@@ -262,9 +262,10 @@ export default class MentionsTextInput extends Component {
       return;
     }
 
+    const nameRegex = this.props.nameRegex ? this.props.nameRegex : `[a-zA-Z0-9_-]*`;
```

Comment:
since there is no string interpolation intended in the pattern string, it should probably just be an ordinary string 
`"[a-zA-Z0-9_-]*"`

Note that regular expressions can be created directly using /'s (e.g. `/[A-Za-z0-9_-]*/` but that would not work here since this string ends up being concatenated to another string before creating the full regular expression.